overlord/snapstate/snapmgr.go

```diff
@@ -465,6 +463,23 @@ func (snapst *SnapState) CurrentSideInfo() *snap.SideInfo {
 	panic("cannot find snapst.Current in the snapst.Sequence.Revisions")
 }
 
+// CurrentComponentSideInfos returns the component side infos for the revision
+// indicated by snapst.Current in the snap revision sequence, if there is one.
+func (snapst *SnapState) CurrentComponentSideInfos() []*snap.ComponentSideInfo {
+	if !snapst.IsInstalled() {
+		return nil
+	}
+	if idx := snapst.LastIndex(snapst.Current); idx >= 0 {
+		comps := snapst.Sequence.Revisions[idx].Components
+		csis := make([]*snap.ComponentSideInfo, 0, len(comps))
+		for _, comp := range comps {
+			csis = append(csis, comp.SideInfo)
+		}
+		return csis
```

Comment:
don't we have a helper that does some of this in Sequence?

Reply:
There is `SnapState.Sequence.ComponentsForRevision`, but that returns `[]*sequence.ComponentState` rather than `[]*snap.ComponentSideInfo`.

Reply:
can't we use it here though perhaps?

Reply:
is this the only place that does this?

Reply:
Ah my bad I misunderstood. Changed now.